src/languageservice/services/yamlCompletion.ts

```diff
@@ -1251,25 +1237,32 @@ export class YamlCompletion {
     separatorAfter: string,
     collector: CompletionsCollector,
     types: unknown,
+    completionType: 'property' | 'value',
     isArray?: boolean
   ): void {
     if (typeof schema === 'object') {
       this.addEnumValueCompletions(schema, separatorAfter, collector, isArray);
       this.addDefaultValueCompletions(schema, separatorAfter, collector);
       this.collectTypes(schema, types);
+
+      if (isArray && completionType === 'value' && !isAnyOfAllOfOneOfType(schema)) {
+        // add array only for final types (no anyOf, allOf, oneOf)
+        this.addArrayItemValueCompletion(schema, separatorAfter, collector);
```

Comment:
moving array item value completion into this `addSchemaValueCompletions` has two advantages:
 - we can remove 3 duplicity implementation
 - we can use recursion to `flat` the schema structure